core/domain/acl_decorators.py

```diff
@@ -72,40 +71,286 @@ def check_activity_accessible(
+    """Checks whether the user can unpublish given collection.
+
+    Args:
+        user_id: str. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        collection_id: str. Id of given collection.
+
+    Returns:
+        bool. Whether the user an unpublish given collection.
+    """
+    if not user_id:
+        return False
+
+    collection_rights = rights_manager.get_collection_rights(collection_id)
+    if collection_rights is None:
+        return False
+
+    if (collection_rights.is_published() and
+            role_services.ACTION_UNPUBLISH_PUBLIC_COLLECTION in user_actions):
+        return True
+
+    return False
+
+
+def check_can_delete_exploration(user_id, user_actions, exploration_id):
+    """Checks whether the user can delete given exploration.
+
+    Args:
+        user_id: str. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_id: str. Id of given exploration.
+
+    Returns:
+        bool. Whether the user can delete given exploration.
+    """
+    if not user_id:
+        return False
+
     exploration_rights = rights_manager.get_exploration_rights(
         exploration_id, strict=False)
 
     if exploration_rights is None:
-        raise base.UserFacingExceptions.PageNotFoundException
+        return False
 
-    if exploration_rights.community_owned:
+    if ((exploration_rights.status == (
+            rights_manager.ACTIVITY_STATUS_PRIVATE)) and
+            (role_services.ACTION_DELETE_OWNED_EXPLORATION in (
+                user_actions)) and
+            exploration_rights.is_owner(user_id)):
+        return True
+    elif (exploration_rights.is_published() and
+          role_services.ACTION_DELETE_ANY_PUBLIC_EXPLORATION in (
+              user_actions)):
         return True
 
-    if role_services.ACTION_EDIT_ANY_EXPLORATION in user_actions:
+
+def check_can_modify_exploration_roles(user_id, user_actions, exploration_id):
+    """Checks whether the user can modify roles for given exploration.
+
+    Args:
+        user_id: str. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_id: str. Id of given exploration.
+
+    Returns:
+        bool. Whether the user can modify roles for given exploration.
+    """
+    if not user_id:
+        return False
+
+    exploration_rights = rights_manager.get_exploration_rights(
+        exploration_id, strict=False)
+
+    if exploration_rights is None:
+        return False
+
+    if (exploration_rights.community_owned or
+            exploration_rights.cloned_from):
+        return False
+
+    if (role_services.ACTION_MODIFY_ROLES_FOR_ANY_EXPLORATION in
+            user_actions):
         return True
+    if (role_services.ACTION_MODIFY_ROLES_FOR_OWNED_EXPLORATION in
+            user_actions):
+        if exploration_rights.is_owner(user_id):
+            return True
+
```

Comment:
What's really happening is that None is being returned, which is False-y. But you shouldn't be relying on this sort of implicit behaviour, since the function is supposed to return a boolean anyway and this breaks its contract.